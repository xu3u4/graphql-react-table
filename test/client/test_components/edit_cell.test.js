import React from 'react';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';
import renderer from 'react-test-renderer';

import EditCell from 'components/common/edit_cell';

describe('Render <EditCell>', () => {
  const onChange = sinon.stub();
  let isShowWarning = false;
  let editcell;
  let value = '';

  beforeEach(() => {
    editcell = shallow(
      <EditCell onInput={ onChange } isShowWarning={ isShowWarning }>{ value }</EditCell>
    );
  });

  it('should have <input> inside <td>', () => {
    editcell = renderer.create(
      <EditCell onInput={ onChange } isShowWarning={ isShowWarning }>{ value }</EditCell>
    );
    expect(editcell).toMatchSnapshot();
  });

  it('should call onChange when input', () => {
    editcell.find('input').simulate('change');
    expect(onChange.calledOnce).toBeTruthy;
  });

  describe('if value is set', () => {
    value = 'cat1';

    it('input should have default value', () => {
      expect(editcell.find('input').prop('value')).toEqual('cat1');
    });
  });
});