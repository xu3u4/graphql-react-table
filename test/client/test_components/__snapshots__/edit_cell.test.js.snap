// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <EditCell> should have <input> inside <td> 1`] = `
<td
  className="editCell"
>
  <input
    className="input_row null"
    onChange={[Function]}
    value="cat1"
  />
  <span
    className="hideMessage"
  >
    *Required
  </span>
</td>
`;
